cmd/util/ledger/migrations/deduplicate_contract_names_migration.go

```diff
@@ -0,0 +1,111 @@
+package migrations
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/fxamacker/cbor/v2"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/cadence/runtime/common"
+
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/convert"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// DeduplicateContractNamesMigration checks if the contract names have been duplicated and
+// removes the duplicate ones.
+type DeduplicateContractNamesMigration struct {
+	log zerolog.Logger
+}
+
+func (d *DeduplicateContractNamesMigration) Close() error {
+	return nil
+}
+
+func (d *DeduplicateContractNamesMigration) InitMigration(
+	log zerolog.Logger,
+	_ []*ledger.Payload,
+	_ int,
+) error {
+	d.log = log.
+		With().
+		Str("migration", "DeduplicateContractNamesMigration").
+		Logger()
+
+	return nil
+}
+
+func (d *DeduplicateContractNamesMigration) MigrateAccount(
+	ctx context.Context,
+	address common.Address,
+	payloads []*ledger.Payload,
+) ([]*ledger.Payload, error) {
+	flowAddress := flow.ConvertAddress(address)
+	contractNamesID := flow.ContractNamesRegisterID(flowAddress)
+
+	var contractNamesPayload *ledger.Payload
+	contractNamesPayloadIndex := 0
+	for i, payload := range payloads {
+		key, err := payload.Key()
+		if err != nil {
+			return nil, err
+		}
+		id, err := convert.LedgerKeyToRegisterID(key)
+		if err != nil {
+			return nil, err
+		}
+		if id == contractNamesID {
+			contractNamesPayload = payload
+			contractNamesPayloadIndex = i
+			break
+		}
+	}
+	if contractNamesPayload == nil {
+		return payloads, nil
+	}
+
+	var contractNames []string
+	err := cbor.Unmarshal(contractNamesPayload.Value(), &contractNames)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get contract names: %w", err)
+	}
+
+	var foundDuplicate bool
```

Comment:
We need to sort `contractNames` before iterating contract names for duplicates.

Reply:
contract names must be sorted by definition, so there is no need.

Reply:
I see.  Can we add a comment that contract names are already sorted at this point?

Also it would be good to verify contract names being sorted in tests.

Reply:
added verification and tests